runtime/io/jvm/src/aws/smithy/kotlin/runtime/io/RandomAccessFileSource.kt

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package aws.smithy.kotlin.runtime.io
+
+import okio.Buffer
+import okio.Timeout
+import java.io.File
+import java.io.IOException
+import java.io.RandomAccessFile
+import kotlin.jvm.Throws
+
+internal class RandomAccessFileSource(
+    file: File,
+    start: Long,
+    private val endInclusive: Long,
+    private val timeout: Timeout = Timeout.NONE,
+) : okio.Source {
```

Comment:
**Question**: Doesn't Okio already provide a file source?

Reply:
It does but not one that can consume arbitrary byte ranges

Reply:
Ah, too bad.